cvat/apps/dataset_manager/annotation.py

```diff
@@ -169,29 +169,43 @@
         self.tracks = []
 
 class AnnotationManager:
-    def __init__(self, data):
+    def __init__(self, data: AnnotationIR, *, dimension: DimensionType):
         self.data = data
+        self.dimension = dimension
+
+    def merge(self, data: AnnotationIR, start_frame: int, overlap: int):
+        tags = TagManager(self.data.tags, dimension=self.dimension)
+        tags.merge(data.tags, start_frame, overlap)
+
+        shapes = ShapeManager(self.data.shapes, dimension=self.dimension)
+        shapes.merge(data.shapes, start_frame, overlap)
+
+        tracks = TrackManager(self.data.tracks, dimension=self.dimension)
+        tracks.merge(data.tracks, start_frame, overlap)
+
+    def clear_frames(self, frames: Container[int]):
+        if not isinstance(frames, set):
+            frames = set(frames)
 
-    def merge(self, data, start_frame, overlap, dimension):
-        tags = TagManager(self.data.tags)
-        tags.merge(data.tags, start_frame, overlap, dimension)
+        tags = TagManager(self.data.tags, dimension=self.dimension)
+        tags.clear_frames(frames)
 
-        shapes = ShapeManager(self.data.shapes)
-        shapes.merge(data.shapes, start_frame, overlap, dimension)
+        shapes = ShapeManager(self.data.shapes, dimension=self.dimension)
+        shapes.clear_frames(frames)
 
-        tracks = TrackManager(self.data.tracks, dimension)
-        tracks.merge(data.tracks, start_frame, overlap, dimension)
+        if self.data.tracks:
+            # Tracks are not expected in the cases this function is supposed to be used
+            raise AssertionError("Partial annotation cleanup is not supported for tracks")
```

Comment:
`NotImplementedError` would mean that it might be implemented in the future. It's not the case here, `AssertionError` indicates that if the function is called for tracks, it's the caller (the developer) error.